nomad/structs/structs.go

```diff
@@ -11020,6 +11032,19 @@ func (a *Allocation) SupportsDisconnectedClients(serverSupportsDisconnectedClien
 	return false
 }
 
+// AvoidRescheduleOnLost determines if an alloc allows to have a replacement
+// when lost.
+func (a *Allocation) AvoidRescheduleOnLost() bool {
+	if a.Job != nil {
+		tg := a.Job.LookupTaskGroup(a.TaskGroup)
+		if tg != nil {
+			return tg.AvoidRescheduleOnLost
+		}
+	}
+
+	return false
```

Comment:
Have you hit a `panic` where `a.Job` was `nil`? 

We denormalize the allocation in some parts of the scheduler and remove `Job` from the allocation. I suspect this may be the case for `isValidForLostNode()` because it acts on the plan, so we may be incorrectly returning `false` here.

Reply:
I have not seen it yet, it returns the default value, im not sure what to do if the alloc does not have the job